src/wallet/wallet.cpp

```diff
@@ -2501,6 +2501,13 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
     wtxNew.fTimeReceivedIsTxTime = true;
     wtxNew.BindWallet(this);
     CMutableTransaction txNew;
+    // This transaction is used to track the best transaction that has change
+    // falling between uneconomical dust and MIN_FINAL_CHANGE. If the wallet
+    // runs out of funds trying to find a transaction that has pruneable
+    // change dust or change larger than MIN_FINAL_CHANGE, it will
+    // return this transaction.
+    CMutableTransaction tx_cached;
+    bool have_cached_txn;
```

Comment:
Shouldn't you assign `have_cached_txn` to false here as it is not assiged before it is used down below?